JASP-Engine/JASP/R/principalcomponentanalysis.R

```diff
@@ -745,31 +747,87 @@ PrincipalComponentAnalysis <- function(dataset = NULL, options, perform = "run",
     list(name = "p", title = "p", type="number", format = "dp:3;p:.001")
     # list(name = "RMS", title = "RMS", type="number", format = "dp:3")
   ))
-  
+
+  # Extract loadings:
+  if(is.null(analysisResults) || isTryError(analysisResults)){# || perform == "init"
+    Fits <- list(
+      CHI = ".",
+      PVAL = ".",
+      DF = "."
+    )
+
+    if(isTryError(analysisResults)){
+      errorMessage <- .extractErrorMessage(analysisResults)
+      .addFootnote(footnotes, symbol="<em>Error.</em>", text=errorMessage)
+    }
+  }else{
+    analysisResults <- analysisResults$Results
+
+    Fits <- list(
+      CHI = .DotIfNULL(analysisResults$STATISTIC),
+      PVAL = .DotIfNULL(analysisResults$PVAL),
+      DF = .DotIfNULL(analysisResults$dof)
+    )
+  }
+
   # Create and fill the row(s):
   goodnessOfFit[["data"]] <- list(
     list(
       model = "Model", 
       chisq = ifelse(Fits$DF>0,Fits$CHI,"."),
       df = ifelse(Fits$DF>0,Fits$DF,"."),
       p = ifelse(Fits$DF>0,Fits$PVAL,".")
-      # RMS = Fits$RMS
     )
   )
+
+  if(Fits$DF <= 0) .addFootnote(footnotes, symbol="<em>Warning.</em>", text="Degree of freedom is not a valid number.")
   
```

Comment:
@TimKDJ how do deal with warnings?

Reply:
warning removed, not sure how to present it properly